brave/brave-core
# NFT pinning optimisation

Resolves https://github.com/brave/brave-browser/issues/29230
Before: ipfs://bafy/1/1.png is given and ipfs://bafy is pinned recursively.
After: bafy and bafy/1 pinned non-recursively and bafy/1/1.png pinned recursively.

Resolves 



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
0) Overall test that NFT pinning works.
1) Pin some NFT which has metadata or image format like ipfs.io/ipfs/CID/../*.jpg(json,png etc).
2) Check whether CID is pinned using direct mode by calling curl -X POST http://127.0.0.1:45001/api/v0/pin/ls?arg=CID
3) Check whether image\metadata of the NFT are pinned recursively by calling
http://127.0.0.1:45001/api/v0/pin/ls?arg=%2Fipfs%2F<CID>%2F<PATH>
